save-sandbox/src/main/kotlin/com/saveourtool/save/sandbox/repository/SandboxUserRepository.kt

```diff
@@ -23,5 +24,58 @@ class SandboxUserRepository(
                 mapOf("name" to name),
                 Long::class.java
             )
-                .orNotFound { "There is not user with name $name" }
+                .orNotFound { "There is no user with name $name" }
+
+    /**
+     * @param name
+     * @return user or null if no results have been found
+     */
+    @Transactional
+    fun findByName(name: String): User? {
+        val record = namedParameterJdbcTemplate.queryForList(
+            "SELECT * FROM save_cloud.user WHERE name = :name",
+            mapOf("name" to name)
+        ).single()
+            .orNotFound {
+                "There is no user with name $name"
+            }
+
+        return record.toUserEntity()
+    }
+
+    /**
+     * @param name
+     * @param source
+     * @return user or null if no results have been found
+     */
+    fun findByNameAndSource(name: String, source: String): User? {
+        val record = namedParameterJdbcTemplate.queryForList(
+            "SELECT * FROM save_cloud.user WHERE name = :name AND source = :source",
+            mapOf("name" to name, "source" to source),
+        ).single()
+            .orNotFound {
+                "There is no user with name $name and source $source"
+            }
+        return record.toUserEntity()
+    }
+
+    private fun Map<String, Any>.toUserEntity(): User {
+        val record = this
+        return User(
+            name = record["name"] as String?,
```

Comment:
Can we use spring data instead of raw records here?

Reply:
Could you please describe more detailly, what do you mean?

Reply:
can we use
```
return namedParameterJdbcTemplate.queryForList(
            "SELECT * FROM save_cloud.user WHERE name = :name",
            mapOf("name" to name),
            User::class.java
        ).singleOrNull().orNotFound {
                "There is no user with name $name"
            }
```
without manual converting ResultSet to entity?

Reply:
I experimented with it, but didn't found any satisfying option, except to operate with `ResultSet` directly

`queryForList` and `queryForMap` returns the maps in the form `{"key": "value", "key2": "value2", ..... }`, attempts to deserialize it with help of something like
`User::class.java` or `List<User::class.java>` gives no results - it failed on stage with DB communication, without any informative messages